src/main/java/uk/gov/hmcts/ccd/domain/service/DocLinksRestoreService.java

```diff
@@ -0,0 +1,133 @@
+        Query query = jurisdictionList.isEmpty() ? em.createNativeQuery(OLD_CASE_QUERY, CaseDetailsEntity.class) :
+            em.createNativeQuery(OLD_CASE_QUERY_WITH_JUIDS, CaseDetailsEntity.class).setParameter("jids", jurisdictionList);
+        return findInOldCasesBasedOnEventHistory(query.getResultList());
+    }
+
+    private List<CaseDetailsEntity> findDocLinksMissedNewCases(List<String> jurisdictionList) {
+        Query query = jurisdictionList.isEmpty() ? em.createNativeQuery(NEW_CASE_QUERY, CaseDetailsEntity.class) :
+            em.createNativeQuery(NEW_CASE_QUERY_WITH_JUIDS, CaseDetailsEntity.class).setParameter("jids", jurisdictionList);
+        return findInNewCasesBasedOnEventHistory(query.getResultList());
+    }
+
+    private List<CaseDetailsEntity> findInOldCasesBasedOnEventHistory(List<CaseDetailsEntity> caseDetailsEntities) {
+        Map<Long, CaseDetailsEntity> caseMap = getCaseIdMap(caseDetailsEntities);
+
+        Query query = em.createNativeQuery(LAST_EVENT_WITH_VALID_DATA, CaseAuditEventEntity.class).setParameter("caseIds", caseMap.keySet());
+        List<CaseAuditEventEntity> lastWellKnownEvents = query.getResultList();
+
+        Map<CaseDetailsEntity, CaseAuditEventEntity> caseToEventsMap = lastWellKnownEvents.stream()
+            .collect(Collectors.toMap(event -> caseMap.get(event.getCaseDataId()), Function.identity()));
+
+        return caseToEventsMap.keySet().stream()
+            .filter(key -> hasMissingDocuments(key, caseToEventsMap.get(key)))
+            .collect(Collectors.toList());
+    }
+
+    private List<CaseDetailsEntity> findInNewCasesBasedOnEventHistory(List<CaseDetailsEntity> caseDetailsEntities) {
+        Map<Long, CaseDetailsEntity> caseMap = getCaseIdMap(caseDetailsEntities);
+
+        Query query = em.createNativeQuery(NEW_CASE_EVENTS, CaseAuditEventEntity.class).setParameter("caseIds", caseMap.keySet());
+        List<CaseAuditEventEntity> caseEvents = query.getResultList();
+
+        Map<Long, List<CaseAuditEventEntity>> eventsByCaseIdMap = caseEvents.stream().collect(groupingBy(CaseAuditEventEntity::getCaseDataId));
+
+        Map<CaseDetailsEntity, List<CaseAuditEventEntity>> caseToEventsMap = caseDetailsEntities.stream()
+            .collect(Collectors.toMap(Function.identity(), caseData -> eventsByCaseIdMap.get(caseData.getId())));
+
+        return caseToEventsMap.keySet().stream()
+            .filter(key -> hasMissingDocuments(key, caseToEventsMap.get(key)))
+            .collect(Collectors.toList());
+    }
+
+    private Map<Long, CaseDetailsEntity> getCaseIdMap(List<CaseDetailsEntity> caseDetailsEntities) {
+        return caseDetailsEntities.stream()
+                .collect(Collectors.toMap(CaseDetailsEntity::getId, Function.identity()));
+    }
+
+    private boolean hasMissingDocuments(CaseDetailsEntity caseDetails, CaseAuditEventEntity caseEvent) {
+        List<String> eventDocValues = caseEvent.getData().findValuesAsText(DOCUMENT_URL);
```

Comment:
we are not distinguishing between a doc in a collection and a doc is a simple field. As a result we might be classifying as 'Needs Review by User' more cases than we should. Because users can only delete themselves from collections. So if a case has got all non collections doc fields, and is recovered, the recovery is correct without doubts. We should not mark the case as 'Needs Review by User'. I think it's worth improving the program to do that

Reply:
I guess we could add this check as part of the Phase 2 and ignore it here.

Reply:
The current logic correctly identifies impacted case irrespective of where the doc links are missing. So, we need a classification here something like `impacted but not required a user review` . 
First of all, are we sure docLinks in a simple filed actually missing because of this bug? I mean anyone verified locally?

Reply:
Yes agree we need a classification `impacted but not required a user review`. And yes simple document fields are impacted too.